### DISCUSSION TITLE
2.6.0

### DISCUSSION THREAD

**Discussion by ruibaby** [halo-dev/halo#4022] [General]
## 新特性

- 新增文章和自定义页面预览功能。 by @guqing in <https://github.com/halo-dev/halo/pull/3983>
- 支持通过 URI 远程安装和升级插件。 by @guqing in <https://github.com/halo-dev/halo/pull/3963>
- 支持通过 URI 远程安装和升级主题。 by @guqing in <https://github.com/halo-dev/halo/pull/3939>

## 功能优化

- 修改 Console 的路由模式为 HTML5 模式。 by @ruibaby in <https://github.com/halo-dev/halo/pull/4005>
- 优化插件安装失败的提示信息。 by @guqing in <https://github.com/halo-dev/halo/pull/3993>
- Console 端的登录页面支持登录之后重定向到某个站内链接。 by @ruibaby in <https://github.com/halo-dev/halo/pull/3989>
- 优化 Console 端文章分类和标签选择器通过关键词搜索之后选择的逻辑。 by @zjy4fun in <https://github.com/halo-dev/halo/pull/3978>
- Console 端支持安装已存在主题时提示升级主题。 by @ruibaby in <https://github.com/halo-dev/halo/pull/3970>
- 重构排序参数以统一自定义 API 的风格。 by @guqing in <https://github.com/halo-dev/halo/pull/3956>
- 重构 Console 插件管理的异常状态显示。 by @ruibaby in <https://github.com/halo-dev/halo/pull/3945>
- 对客户端评论及回复列表接口进行脱敏处理。 by @LIlGG in <https://github.com/halo-dev/halo/pull/3936>
- 优化 Console 端部分数据列表的可访问性。 by @ruibaby in <https://github.com/halo-dev/halo/pull/3919>
- 支持系统初始化检查，如果未初始化则重定向到控制台。 by @guqing in <https://github.com/halo-dev/halo/pull/3892>

## 问题修复

- 修复长时间删除插件失败的问题。 by @JohnNiang in <https://github.com/halo-dev/halo/pull/4002>
- 修复 Console 端 Dialog 对话框组件中的文字内容可能会溢出容器的问题。 by @ruibaby in <https://github.com/halo-dev/halo/pull/4007>
- 修复页面别名包含特殊字符导致无法访问的问题。 by @JohnNiang in <https://github.com/halo-dev/halo/pull/4013>
- 修复 Console 端插件详情中的协议出现两次相同内容的问题。 by @zjy4fun in <https://github.com/halo-dev/halo/pull/3977>
- 修复 Console 端认证方式列表无法使用搜索的问题。 by @JustinLiang522 in <https://github.com/halo-dev/halo/pull/3974>
- 修复没有主题管理权限时，进入主题功能页面报错的问题。 by @LIlGG in <https://github.com/halo-dev/halo/pull/3938>
- 修复在 Windows 系统无法启动插件的问题。 by @guqing in <https://github.com/halo-dev/halo/pull/3925>
- 修复主题模板中 site.url 解析有误的问题。 by @JohnNiang in <https://github.com/halo-dev/halo/pull/3910>
- 解决访问页面无法渲染的问题。 by @JohnNiang in <https://github.com/halo-dev/halo/pull/3909>
- 修复在初始化时，预设插件可能初始化失败的问题。 by @ruibaby in <https://github.com/halo-dev/halo/pull/3894>
- 修复删除某个插件项导致其他插件 Logo 渲染失败的问题。 by @zjy4fun in <https://github.com/halo-dev/halo/pull/3888>
- 修复创建用户或者修改用户资料时，头像链接不支持相对链接的问题。 by @ruibaby in <https://github.com/halo-dev/halo/pull/3880>
- 修复部分 FormKit 的 radio 表单项的 label 设置不正确的问题。 by @ruibaby in <https://github.com/halo-dev/halo/pull/3878>
- 修复依然能搜索到已删除文章的问题。 by @JohnNiang in <https://github.com/halo-dev/halo/pull/3877>

## 依赖更新

- 升级 Spring Boot 至 3.1.0。 by @JohnNiang in <https://github.com/halo-dev/halo/pull/3996>
- 替换 R2DBC MySQL 驱动为 `io.asyncer:r2dbc-mysql`。 by @JohnNiang in <https://github.com/halo-dev/halo/pull/3918>

## 开发者相关

- Console 端的 FormKit Repeater 输入框支持 min 和 max 参数用于限制项目数量。 by @ruibaby in <https://github.com/halo-dev/halo/pull/3965>

## 文档

<https://docs.halo.run>

## 资源

适用于 2.0 的插件和主题可查阅：

- <https://halo.run/store/apps>
- <https://github.com/halo-sigs/awesome-halo>

**Full Changelog**: <https://github.com/halo-dev/halo/compare/v2.5.0...v2.6.0>

<hr /><em>This discussion was created from the release <a href='https://github.com/halo-dev/halo/releases/tag/v2.6.0'>2.6.0</a>.</em>

**Comment by mobeicanyue** [comment]
想请教一下 咱们action自动构建的release 是怎么编辑上面的那堆新特性 修复等一大段文字的，是release之后手动编辑的吗

**Reply by JohnNiang** [reply]
大部分是手动编辑的。

如果你想自动化 Release Notes，这里推荐使用 <https://github.com/release-drafter/release-drafter>。

**Reply by ruibaby** [reply]
并不是完全手动编辑的，目前我们的 PR 会在合并前编辑好 Release Note 和设置 Milestone，最终发布的时候目前是使用 GitHub 的 API 来获取某个 Milestone 下的所有 PR 并拿到 Release Note。

```ts
import { Octokit } from "octokit";
import { marked } from "marked";
import { load } from "cheerio";

const octokit = new Octokit({
  auth: "<PRIVATE>",
});

const repo = "halo";
const milestone = "2.6.0";

async function run() {
  const data = await octokit.request("GET /repos/{owner}/{repo}/pulls", {
    owner: "halo-dev",
    repo: repo,
    headers: {
      "X-GitHub-Api-Version": "2022-11-28",
    },
    state: "closed",
    per_page: 100,
    page: 1,
  });

  const latestPRs = data.data.filter((pr: any) => {
    return pr.milestone?.title === milestone;
  });

  latestPRs.forEach((pr: any) => {
    const $ = load(marked.parse(pr.body));
    const note = $(".language-release-note").text().trim();
    console.log(`- ${note} by @${pr.user.login} in <${pr.html_url}>`);
  });
}

run();
```
